src/home/homed-home.c

```diff
@@ -2759,7 +2759,7 @@ int home_create_fifo(Home *h, bool please_suspend) {
                 if (mkfifo(fn, 0600) < 0 && errno != EEXIST)
                         return log_error_errno(errno, "Failed to create FIFO %s: %m", fn);
 
-                ref_fd = open(fn, O_RDONLY|O_CLOEXEC|O_NONBLOCK);
+                ref_fd = open(fn, O_PATH|O_CLOEXEC|O_NONBLOCK);
```

Comment:
similar

Reply:
@poettering what do you think about adding two functions named like `check_fifo_{read,writ}able()` which do this?
I could take care of this.